.github/workflows/C++.yml

```diff
@@ -0,0 +1,79 @@
+name: C++ Build
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  Linux:
+    runs-on: ubuntu-latest
+    container: quay.io/pypa/manylinux2014_x86_64:latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install fftw
+      run: |
+        yum install -y fftw3-devel
+
+    - name: Compile C++ code
+      run: |
+        make lib
+        make spreadtestall
+        make test
+
+  MacOS_clang:
+    runs-on: macos-latest
+    env:
+      MACOSX_DEPLOYMENT_TARGET: 10.15
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install omp and fftw
+      run: |
+        brew install libomp fftw
+        cp make.inc.macosx_clang make.inc
+
+    - name: Compile C++ code
+      run: |
+        make lib
+        make spreadtestall
+        make test
+
+  MacOS_gcc:
+    runs-on: macos-latest
+    env:
+      MACOSX_DEPLOYMENT_TARGET: 10.15
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install gcc and fftw
+      run: |
+        brew install gcc@10 fftw
+        cp make.inc.macosx_gcc-10 make.inc
+
+    - name: Compile C++ code
+      run: |
+        make lib
+        make spreadtestall
+        make test
+
+  Windows:
```

Comment:
Can you describe in comments what tricks are used for windows system here (msys not mingw, etc) ? For future devs to understand.

Reply:
Yes, it's confusing. Adding comments in makefile.

I am using mingw here.
mingw is the gcc compiler for windows system, and the mingw compiler can be used in both command line environments, i.e., msys or wsl.
In wsl, 'copy' or 'move' commands in makefile is recognized; in msys, 'copy' or 'move' are not recognized, I have to use 'cp' or 'mv' as in linux/osx cases.